jscomp/frontend/bs_builtin_ppx.ml

```diff
@@ -224,9 +224,20 @@ let expr_mapper ~async_context ~in_function_def (self : mapper)
   match Ast_attributes.has_await_payload e.pexp_attributes with
   | None -> result
   | Some _ ->
-    if !async_context = false then
-      Location.raise_errorf ~loc:e.pexp_loc
-        "Await on expression not in an async context";
+    (if !async_context = false then
+     let isJsImport (e : Parsetree.expression) =
+       match e with
+       | {
+        pexp_desc =
+          Pexp_apply
+            ({pexp_desc = Pexp_ident {txt = Ldot (Lident "Js", "import")}}, _);
+       } ->
+         true
+       | _ -> false
+     in
+     if not (isJsImport e) then
```

Comment:
This completely disables the check for async context in the case of `Js.import`.
That will generate invalid code when e.g. `Js.import` is used in a function that is not async.

Reply:
Same comment as above: Dynamic import is allowed in the top-level without async context.

Reply:
Try to write some tests of obviously wrong code, and it will be clear why you can't just turn all the checks off.

Reply:
Correct. I've restored it
```rescript
let f = () => {
  await Js.import(Belt.List.forEach)
}
```
This should be a type error.